extensions/positron-python/python_files/posit/positron/jedi.py

```diff
@@ -107,19 +117,121 @@ def _interpreter_goto(
     ]
 
 
+def _completion_api_complete(self: CompletionAPI) -> List[Completion]:
+    leaf = self._module_node.get_leaf_for_position(self._original_position, include_prefixes=True)
+    string, start_leaf, quote = completion_api._extract_string_while_in_string(  # noqa: SLF001
+        leaf, self._original_position
+    )
+
+    # Try to complete `os.getenv` calls.
+    if string is not None and quote is not None:
+        completions = _complete_os_getenv(
+            start_leaf,
+            self._inference_state,
+            quote + string,
+            self._signatures_callback,
+            self._code_lines,
+            self._original_position,
+            self._fuzzy,
+        )
+        if completions:
+            return completions
+
+    return _original_completion_api_complete(self)
+
+
+def _complete_os_getenv(
+    start_leaf,
+    inference_state: InferenceState,
+    string: str,
+    signatures_callback,
+    code_lines,
+    position,
+    fuzzy: bool,  # noqa: FBT001
+):
+    def complete():
+        # As a shortcut, create a compiled value for `os.environ` and use its dict completions.
+
+        # Create the utility strings.
+        cut_end_quote = ""
+        if string:
+            cut_end_quote = strings.get_quote_ending(
+                string, code_lines, position, invert_result=True
+            )
+
+        # Create the compiled value for `os.environ`.
+        access_path = create_access_path(inference_state, os.environ)
+        compiled_value = create_from_access_path(inference_state, access_path)
+        dicts = ValueSet([_OsEnvironCompiledValueWrapper(compiled_value)])
+
+        # Return the completions.
+        return list(_completions_for_dicts(inference_state, dicts, string, cut_end_quote, fuzzy))
+
+    # See `jedi.api.file_name._add_os_path_join` for a useful reference with similar behavior.
+
+    # Are we completing an `os.getenv` call?
+    signatures = signatures_callback(*position)
+    if signatures and all(s.full_name == "os.getenv" for s in signatures):
```

Comment:
Testing this locally: this always works when I autocomplete `os.getenv("`. 🎉 

For `os.environ["`, it's a little stranger -- it's not working for me unless I actually run `import os` in the console first. Once I do, the autocomplete works. But when I restart the interpreter the autocompletion goes back to only listing files.

Can you reproduce this?

Reply:
Ah good catch! It should work now.

Reply:
Yes, works now! Thanks!